libcextract/FunctionDepsFinder.hh

```diff
@@ -43,6 +43,33 @@ class ClosureSet
     Dependencies.insert(decl);
   }
 
+  /*
+   * Check if the same Decl was already added into the closure but for a
+   * different typename.
+   */
+  TypedefDecl *insideRangeOfDecl(TypedefDecl *decl)
+  {
+    SourceRange range = decl->getSourceRange();
+
+    for (auto it = Dependencies.begin(); it != Dependencies.end(); ++it) {
```

Comment:
This algorithm is O(_n_ ²). If the closure is small, I think this is acceptable, but if we want to do this extensively we may be in a better shape to sort by SourceLocation something to reduce it to O(n log n).

Reply:
The idea in this case was to submit the solution to check with our CI. Also, for my current LP, it worked.

I'll think how this could be improved.